src/fbp/image_geom.jl

```diff
@@ -288,7 +288,7 @@ function _down_round(
 ) where {D}
 	out = val ./ down
 	# for non-divisors make dim a multiple of 2
-	fun = out -> out == round(out) ? out : 2 * round(out / 2)
+	fun = out -> out == round(out) ? out : 2 * (out ÷ 2)
 	out = fun.(out)
 	dd = dd .* down
 	return Int.(out), dd
```

Comment:
Unnecessary intermediate memory allocation can also be a root of performance issue:

```julia
out = rand(Float64, 64, 64, 30);

function f1(out)
    fun = out -> out == round(out) ? out : 2 * (out ÷ 2)
    out = fun.(out) # unnecessary memory allocation
    return Int.(out) # memory allocation
end

function f2(out)
    fun = out -> out == round(out) ? out : 2 * (out ÷ 2)
    # the @ macro will fuse all the operations and thus only allocate memory once
    return @. Int(fun(out))
end
```

```julia
julia> @btime f1($out);
  677.139 μs (4 allocations: 1.88 MiB)

julia> @btime f2($out);
  502.425 μs (2 allocations: 960.08 KiB)
```

Reply:
thanks for this tip too.  in this particular case it is only applied to small tuples, not big arrays, but i will change it anyway to remind me to use `@.` more in the future.  i really appreciate it.